packages/server/lib/util/require_async_child.js

```diff
@@ -24,6 +27,14 @@ function run (ipc, requiredFile, projectRoot) {
     throw new Error('Unexpected: projectRoot should be a string')
   }
 
+  if (!tsRegistered) {
+    debug('register typescript for required file')
+    tsNodeUtil.register(projectRoot, requiredFile)
+
+    // ensure typescript is only registered once
+    tsRegistered = true
+  }
```

Comment:
Why does this need to run if `cypress.config.js` is being used? Would it improve performance by a tangible amount if we only ran `.register()` if `requiredFile.endsWith('.ts')`? Could we avoid unnecessarily `require`ing `typescript` in the child process every time?

Reply:
I added this to allow users to require `ts` files in their `cypress.config.js` file.
I don't know if it is the best. But since this parse is done only once now, the gain is not significant enough to me to make users have to choose.

What do you think?

Reply:
Can we change line 30 to `if (!tsRegistered && requiredFile.endsWith('.ts')) {`? That would alleviate my concern I think
```suggestion
  if (!tsRegistered && requiredFile.endsWith('.ts')) {
    debug('register typescript for required file')
    tsNodeUtil.register(projectRoot, requiredFile)

    // ensure typescript is only registered once
    tsRegistered = true
  }
```

Reply:
Updated in 0ad2ded